src/Common/Exception.cpp

```diff
@@ -429,17 +429,20 @@ PreformattedMessage getCurrentExceptionMessageAndPattern(bool with_stacktrace, b
         }
         catch (...) {}
 
-// #ifdef ABORT_ON_LOGICAL_ERROR
-//         try
-//         {
-//             throw;
-//         }
-//         catch (const std::logic_error &)
-//         {
-//             abortOnFailedAssertion(stream.str());
-//         }
-//         catch (...) {}
-// #endif
+#ifdef ABORT_ON_LOGICAL_ERROR
+        try
+        {
+            throw;
+        }
+        catch (const std::logic_error &)
+        {
+            if (!with_stacktrace)
```

Comment:
Did you mean 
```
if (with_stacktrace)
```

Reply:
No, if with_stacktrace is true, then we already have the stacktrace printed to `stream`. And we need to print it if it's false